libraries/AP_GPS/GPS_Backend.cpp

```diff
@@ -112,3 +114,38 @@ void AP_GPS_Backend::fill_3d_velocity(void)
     state.velocity.z = 0;
     state.have_vertical_velocity = false;
 }
+
+
+void AP_GPS_Backend::_detection_message(char *buffer, const uint8_t buflen) const
+{
+    const uint8_t instance = state.instance;
+    const struct AP_GPS::detect_state dstate = gps.detect_state[instance];
+
+    if (dstate.detect_started_ms > 0) {
```

Comment:
@peterbarker I was referring to these comments: https://github.com/ArduPilot/ardupilot/pull/4584#r77431430

Just to be more clear on this: looking at code it seems to me like the baud will be correct here but just by chance. If we changed the order of the baud rates these 3 drivers would be created and destroyed until the correct baud was found - that would mean that this detected message would also be wrong and repeated multiple times.

I don't want to overcomplicate and I think this PR is better than what we have now. It has been here for quite a while and I don't want it to bit rot, so as soon as @WickedShell confirms this is fine, let's put it in.